pkg/controller/job/job_controller_test.go

```diff
@@ -654,9 +643,14 @@ func TestControllerSyncJob(t *testing.T) {
 			} else if tc.fakeExpectationAtCreation > 0 {
 				manager.expectations.ExpectCreations(key, int(tc.fakeExpectationAtCreation))
 			}
+			originalConditions := []batch.JobCondition{
+				newCondition(batch.JobComplete, v1.ConditionFalse, "", ""),
+				newCondition(batch.JobFailed, v1.ConditionFalse, "", ""),
```

Comment:
also `newCondition(batch.JobSuspended, v1.ConditionFalse, "", "")`, if feature is enabled.

But, I would rather prefer we have `wantConditions []JobCondition` in the test case.